pkg/apis/core/types.go

```diff
@@ -3710,6 +3710,7 @@ type ServiceSpec struct {
 	// The special value "*" may be used to mean "any topology". This catch-all
 	// value, if used, only makes sense as the last value in the list.
 	// If this is not specified or empty, no topology constraints will be applied.
+	// This field is alpha-level and is only honored by servers that enable the ServiceTopology feature.
```

Comment:
should we just deprecate this now to start the clock?

Reply:
Yes, I don't see why not

Reply:
Why should this be deprecated ? Is ServiceTopology feature not being used anymore ?

Reply:
Yes.  It was an alpha design that we decided there are better answers to.  SO we are deprecating the API for it and will eventually remove it.

Reply:
Can you suggest some alternatives to achieve the same effect ? Basically to preferentially route traffic based on the cluster topology ?

Reply:
For posterity, this is being built into kube-proxy as [Topology Aware proxying](https://github.com/kubernetes/enhancements/issues/2030#issuecomment-701056673)